task 1

In [8]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline


task 1 -step 1-2-3

In [61]:
data= pd.read_csv('/Users/sajaz/Downloads/AthleteSelection.csv',index_col=('Athlete'))
print(data.head(5))
names=pd.DataFrame(data)
print(names)
y=data.pop('Selected').values
X=data.values
print(X)
print(y)

         Speed  Agility  Selected
Athlete                          
x1        2.50     6.00         0
x2        3.75     8.00         0
x3        2.25     5.50         0
x4        3.25     8.25         0
x5        2.75     7.50         0
         Speed  Agility  Selected
Athlete                          
x1        2.50     6.00         0
x2        3.75     8.00         0
x3        2.25     5.50         0
x4        3.25     8.25         0
x5        2.75     7.50         0
x6        4.50     5.00         0
x7        3.50     5.25         0
x8        3.00     3.25         0
x9        4.00     4.00         0
x10       4.25     3.75         0
x11       2.00     2.00         0
x12       5.00     2.50         0
x13       8.25     8.50         1
x14       5.75     8.75         1
x15       4.75     6.25         1
x16       5.50     6.75         1
x17       5.25     9.50         1
x18       7.00     4.25         1
x19       7.50     8.00         1
x20       7.25     3.75         1
[[2.5  6.  ]
 

In [67]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.neighbors import NearestNeighbors
from sklearn.metrics import accuracy_score
from sklearn.metrics import recall_score

model = NearestNeighbors(n_neighbors=2, radius=0.4)
model.fit(X,y)

NearestNeighbors(n_neighbors=2, radius=0.4)

In [69]:
estimator = model.get_params()
print(estimator)

{'algorithm': 'auto', 'leaf_size': 30, 'metric': 'minkowski', 'metric_params': None, 'n_jobs': None, 'n_neighbors': 2, 'p': 2, 'radius': 0.4}


In [82]:
x_input=np.array([[3.25,8.25],[0.2,3.3]])
print(x_input)


[[3.25 8.25]
 [0.2  3.3 ]]


In [110]:
model2=KNeighborsClassifier(n_neighbors=3)
model2.fit(X,y)
y_dash=model2.predict(X)
print('     y:',y)
print('y_dash:',y_dash)

accuracy = accuracy_score(y, y_dash)
print(f"Accuracy of KNeighborsClassifier with K=3: {accuracy:.2f}")

     y: [0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1]
y_dash: [0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1]
Accuracy of KNeighborsClassifier with K=3: 1.00


C:\Users\sajaz\anaconda3\lib\site-packages\sklearn\neighbors\_classification.py:228: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  mode, _ = stats.mode(_y[neigh_ind, k], axis=1)


10 - y-dash and y is the same so i think the accuracy is good , because the test was on the same data who was trained

task 2

In [90]:
data2= pd.read_csv('/Users/sajaz/Downloads/AthleteTest.csv',index_col=('Athlete'))
print(data2.head(5))
names2=pd.DataFrame(data2)
print(names2)
y_test=data2.pop('Selected').values
X_test=data2.values
print(X_test)
print(y_test)

         Speed  Agility  Selected
Athlete                          
t1         3.3      8.2         0
t2         4.5      4.5         0
t3         5.5      7.2         0
t4         3.8      8.8         1
t5         5.5      5.2         0
         Speed  Agility  Selected
Athlete                          
t1         3.3      8.2         0
t2         4.5      4.5         0
t3         5.5      7.2         0
t4         3.8      8.8         1
t5         5.5      5.2         0
t6         8.1      7.8         0
t7         7.7      5.2         1
t8         6.1      5.5         0
t9         5.5      6.0         0
t10        6.1      5.5         1
[[3.3 8.2]
 [4.5 4.5]
 [5.5 7.2]
 [3.8 8.8]
 [5.5 5.2]
 [8.1 7.8]
 [7.7 5.2]
 [6.1 5.5]
 [5.5 6. ]
 [6.1 5.5]]
[0 0 0 1 0 0 1 0 0 1]


In [119]:
model3=KNeighborsClassifier(n_neighbors=3)

In [120]:
model3.fit(X_test,y_test)
y_dash2=model3.predict(X_test)
print('     y:',y_test)
print('y_dash:',y_dash2)

accuracy = accuracy_score(y_test, y_dash2)
print(f"Accuracy of KNeighborsClassifier with K=3: {accuracy:.2f}")

     y: [0 0 0 1 0 0 1 0 0 1]
y_dash: [0 0 0 0 0 0 1 0 0 0]
Accuracy of KNeighborsClassifier with K=3: 0.80


C:\Users\sajaz\anaconda3\lib\site-packages\sklearn\neighbors\_classification.py:228: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  mode, _ = stats.mode(_y[neigh_ind, k], axis=1)


In [121]:
from sklearn import preprocessing
from sklearn.neighbors import KNeighborsClassifier
from sklearn.preprocessing import StandardScaler, MinMaxScaler
from sklearn.metrics import accuracy_score

scaler=preprocessing.StandardScaler().fit(X_test)
X_scaled =scaler.transform(X_test)
q_scaled =scaler.transform([q])
model3.fit(X_scaled,y_test)
model3.kneighbors(q_scaled)

model3.fit(X_scaled,y_test)
y_dash3=model3.predict(X_scaled)
print('     y:',y_test)
print('y_dash:',y_dash3)
accuracy = accuracy_score(y_test, y_dash3)
print(f"Accuracy of KNeighborsClassifier with K=3: {accuracy:.2f}")

     y: [0 0 0 1 0 0 1 0 0 1]
y_dash: [0 0 0 0 0 0 1 0 0 0]
Accuracy of KNeighborsClassifier with K=3: 0.80


C:\Users\sajaz\anaconda3\lib\site-packages\sklearn\neighbors\_classification.py:228: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  mode, _ = stats.mode(_y[neigh_ind, k], axis=1)


In [122]:
scaler2=preprocessing.MinMaxScaler().fit(X_test)
X_scaled2 =scaler2.transform(X_test)
q_scaled2 =scaler2.transform([q])
model3.fit(X_scaled2,y_test)
model3.kneighbors(q_scaled2)

model3.fit(X_scaled2,y_test)
y_dash4=model3.predict(X_scaled2)
print('     y:',y_test)
print('y_dash:',y_dash4)
accuracy = accuracy_score(y_test, y_dash4)
print(f"Accuracy of KNeighborsClassifier with K=3: {accuracy:.2f}")

     y: [0 0 0 1 0 0 1 0 0 1]
y_dash: [0 0 0 0 0 0 1 0 0 0]
Accuracy of KNeighborsClassifier with K=3: 0.80


C:\Users\sajaz\anaconda3\lib\site-packages\sklearn\neighbors\_classification.py:228: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  mode, _ = stats.mode(_y[neigh_ind, k], axis=1)


Accuracy of KNeighborsClassifier with K=1: 0.90
Accuracy of KNeighborsClassifier with K=1 and StandardScaler: 0.90
Accuracy of KNeighborsClassifier with K=1 and MinMaxScaler: 0.90
Accuracy of KNeighborsClassifier with K=2: 0.70
Accuracy of KNeighborsClassifier with K=2 and StandardScaler: 0.70
Accuracy of KNeighborsClassifier with K=2 and MinMaxScaler: 0.70
Accuracy of KNeighborsClassifier with K=3: 0.80
Accuracy of KNeighborsClassifier with K=3 and StandardScaler: 0.80
Accuracy of KNeighborsClassifier with K=3 and MinMaxScaler: 0.80
